packages/flame/lib/src/components/component.dart

```diff
@@ -53,6 +54,8 @@ class Component {
   bool get hasChildren => _children?.isNotEmpty ?? false;
   ComponentSet? _children;
 
+  var _mountCompleter = Completer<void>();
```

Comment:
We'll have to somehow add this without initializing it in the component, maybe with a mixin? Because we don't want more objects than necessary initialized inside component since it affects performance a lot.

Reply:
How can a simple instantiation affect the perfomance a lot? 🤔

Reply:
I am not against a mixin though, but I just want to understand how come this would affect perfomance that much

Reply:
Because you have a lot of components in your game since it is the basic building block, that's why the fields that can in `Component` are lazily initialized.

Reply:
Interesting, I am still curious on how this should impact perfomance, I mean we are instantiating a lot of things all the time in flame, like our `toRect` method and so on.

But either way, it will be simple to make this lazy loaded, I will add some changes regarding that

Reply:
Maybe it can live inside the `_LifecycleManager`? That way it can safely disappear once it is no longer needed.

Reply:
That was the suggestion @alestiago had too,I think it sounds like a good idea.

Reply:
I tried adding it to `_LifecycleManager`, but I didn't like very much. When I did, the logic was half on that class and half on the `Component`, for now, I think we should leave it on the Component, wdyt?

Reply:
Interesting, what logic had to stay in the `Component`?

Reply:
The getter for the hasMounted now have to use the isMounted, which lives on the component, that was the bigger logic.

Reply:
If we simply replace `_mountCompleter` -> `lifecycleManager._mountCompleter`, then nothing needs to be moved anywhere. 

In general, the job of the `LifecycleManager` class is to handle the temporary state that needs to be kept briefly while the component goes through lifecycle transitions; so `mountCompleter` seem to fit perfectly there.

Reply:
> If we simply replace _mountCompleter -> lifecycleManager._mountCompleter, then nothing needs to be moved anywhere.

We are using the isMounted method to check if we return the completer future or not, if we move the completer to the lifecycle manager I believe the isMounted should go there too, otherwise we will end up with logic spread across these two classes.

I don't mind moving isMounted there, but then we will end up with two "proxy" methods forwarding methods from the life cycle manager to the component, not sure what is best.